evals/prompts/inline_code_completion_prompts/prod_prompt_v3.py

```diff
@@ -0,0 +1,97 @@
+from evals.eval_types import InlineCodeCompletionPromptGenerator, NotebookState, ChatPromptGenerator
+
+__all__ = ['prod_prompt_v3']
+
+class _ProdPromptV3(InlineCodeCompletionPromptGenerator):
+    prompt_name = "prod_prompt_v3"
+
+    def get_prompt(self, prefix: str, suffix: str, notebook_state: NotebookState) -> str:
+    
+        return f"""You are a code completion assistant that lives inside of JupyterLab. Your job is to predict the rest of the code that the user has started to write.
+
+You're given the current code cell, the user's cursor position, and the variables defined in the notebook. The user's cursor is signified by the symbol <cursor>.
+
+CRITICAL FORMATTING RULES:
+1. If the cursor appears at the end of a line (especially after a comment), ALWAYS start your code with a newline character
```

Comment:
Interesting choice making this an "always" rule. I can see the logic behind it, but it may not always be the case.

For example:

```python
name = "mito"
print(
```

I'd hope for the AI to suggest `name)`, which would be one the same line.

Reply:
Good point. I was aiming for `end of a line` to mean like a complete line, so the `print(` would fall into the second category. But I agree that this is not clearly phased. I'll experiment with a few other options and see if it brings up our eval score.

Reply:
Nice. I updated this language and it improved the eval score an average of 2.3%